### PR TITLE
Course user progress

### PR DESCRIPTION
**Pull request by PaulBredl**
## Description of changes
Integrater chapter progress and course progress
  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
In case of automated tests, the test coverage should be reasonably high, especially on Service classes and other classes with complex logic.
    
## Checklist before requesting a review
- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfills all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: additionalResolvers/course.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ const resolvers: Resolvers = {
         // combines suggestions from the chapters of the course to a single list of suggestions for the
         // whole course
         suggestions: {
+            // HINT use course service filters instead of manually filtering the chapters
```
Comment (myluki2000) [inline additionalResolvers/course.ts:8]: What exactly do you mean with "Hint"? Is it a "TODO"? Because then it should be written as such IMO (TODOs are usually highlighted in the IDE and can also be searched for if you use standardized terminology across the whole project)

File: additionalResolvers/course.ts [hunk 2]
```diff
@@ -23,17 +23,9 @@ const resolvers: Resolvers = {
...
                     }
-
-                    // if chapter doesn't have a suggested start date, suggest it
-                    if(chapter.suggestedStartDate === null) {
-                        return true;
-                    }
-
-                    // if chapter has a suggested start date and it's in the past, suggest it
-                    return Date.parse(chapter.suggestedStartDate) < currentTime;
                 });
```
Comment (myluki2000) [inline additionalResolvers/course.ts:29]: Don't you need to return true at the end of the filter function? (Or just Date.parse(chapter.startDate) > currentTime instead of the if block)

**Review by myluki2000 — CHANGES_REQUESTED** [review]

File: additionalResolvers/course.ts (hunk 1 above)
Comment (PaulBredl) [reply additionalResolvers/course.ts:8]: HINT is also a standard term in the java world. didnt know it wasnt everywhere

Comment (PaulBredl) [reply additionalResolvers/course.ts:8]: It's like a non urgent todo

Comment (myluki2000) [reply additionalResolvers/course.ts:8]: ok maybe i just didn't know about it

**Review by myluki2000 — APPROVED** [review]
